app/src/main/java/com/lightricks/ex1/Common.java

```diff
@@ -35,13 +35,15 @@ public static boolean hasReadContactsPermission(Context context) {
     public static void glideSetImage(Context context, ImageView ivImage, String imageName) {
         int imageId = getImageId(context, imageName);
         if (imageId == 0){
+            // NOT thread safe! Think about a safe way to preform this operation (the default image calculation)...
```

Comment:
Good point, though what reason do we have to believe that multiple threads will access this function simultaneously?

Reply:
What reason do we have to believe that multiple threads won't access this function simultaneously?

When writing a "common"/utilities classes which meant to be shared and reusable the public functions should avoid such assumptions. if you have assumptions then document them + verify them if you can.